src/pages/en/reference/integrations-reference.md

```diff
@@ -112,6 +114,25 @@ A callback function to add a component framework renderer (i.e. React, Vue, Svel
 - `clientEntrypoint` - path to a file that executes on the client whenever your component is used. This is mainly for rendering or hydrating your component with JS.
 - `serverEntrypoint` - path to a file that executes during server-side requests or static builds whenever your component is used. These should render components to static markup, with hooks for hydration where applicable. [React's `renderToString` callback](https://reactjs.org/docs/react-dom-server.html#rendertostring) is a classic example.
 
+#### `injectRoute` option
+**Type:** `({pattern: string, entryPoint: string}) => void;`
+
+A callback function to inject a route from an integration. Injected routes can be `.astro` pages, or `.js` route handlers.
```

Comment:
I wonder if “from an integration” is superfluous here as these are the Integration API docs.

Are these links pointing to the right parts of the docs here @thepassle? I’m a little unclear if a JS route handler is a “non-HTML page” or an SSR thing? (Also, this and the RFC say “`.js` route” but could this also be a `.ts`?

I also pluralised “routes” as this can technically inject multiple using dynamic patterns.

```suggestion
A callback function to inject routes into an Astro project. Injected routes can be [`.astro` pages](https://docs.astro.build/en/core-concepts/astro-pages/) or [`.js` route handlers](https://docs.astro.build/en/core-concepts/astro-pages/#non-html-pages).
```